actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -89,19 +97,31 @@ class ActionCable.Connection
 
     open: ->
       ActionCable.log("WebSocket onopen event")
-      @disconnected = false
-      @subscriptions.reload()
+      # This is in the case that the browser client opens
+      # the connection regardless of whether a protocol was 
+      # negotiated successfully or not.
+      if not @isProtocolSupported()
+        @close()
+      else
+        ActionCable.log("WebSocket using '" + @protocol() + "' subprotocol.")
```

Comment:
I'd move this logging to the initial `open` entry. And prefer CoffeeScript's string interpolation: `"using #{@protocol()}  subprotocol"`